### PR TITLE
[SW-702] Initial implementation of H2OAutoML pipeline step

### PR DESCRIPTION
**Pull request by jakubhava**
Still work in progress, this is really just init phase

### CODE REVIEW

**Comment by jakubhava** [comment]
Almost ready, Scala is passing, in Python, getters and setters are to be done

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+    val aml = new AutoML(Key.make(uid), new Date(), spec)
+    AutoML.startAutoML(aml)
+    // Block until AutoML finishes
+    aml.get()
+    val model = new H2OMOJOModel(ModelSerializationSupport.getMojoData(aml.leader()))
+    model.setConvertUnknownCategoricalLevelsToNa(true)
+    model
+  }
+
+  private def split(fr: H2OFrame, hc: H2OContext): Array[Key[Frame]] = {
```
Comment (mmalohlava) [inline ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala]: Do we need to have split duplicated here? It is already implemented in core/src/main/scala/water/support/H2OFrameSupport.scala

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+    val trainKey = Key.make[Frame]("train")
```
Comment (mmalohlava) [inline ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala]: What if `train` key already exist?

Comment (jakubhava) [reply ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala]: Good, point, thanks. I will fix this before the release

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala (hunk 1 above)
Comment (jakubhava) [reply ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala]: I will see if I can reuse the code, thanks!

**Comment by jakubhava** [comment]
Yup, I just wanted to quickly test it and finally have it out. I can parametrize the ham_or_spam scripts in both Scala and Python to be parametrized. Will do this before the release as well. Thanks for the review!

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala (hunk 1 above)
Comment (jakubhava) [reply ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala]: Replaced also as part of https://github.com/h2oai/sparkling-water/pull/586

**Pull request merged**